### PR TITLE
Introduce willChangeTransform as a prop for Mask.js

### PR DESCRIPTION
**Pull request by aerozen3**
Introduce new boolean prop willChangeTransform for Mask.js which allows for toggling on and off `will-change: transform` css

## TODO

- [x] Documentation
- [x] Tests
- [ ] Experimental evidence (required for Masonry changes)
- [ ] Accessibility checkup

### CODE REVIEW

**Comment by aerozen3** [comment]
This PR attempts to fix a Safari full screen video only issue.  See the screenshots here

with `will-change: transform`:with will change transform

without `will-change: transform`:no will change transform

File: packages/gestalt/src/__snapshots__/Avatar.test.js.snap [hunk 1]
```diff
@@ -110,7 +110,7 @@ exports[`Avatar renders the Pinterest P on verified 1`] = `
   }
 >
   <div
-    className="Mask circle"
+    className="Mask circle hintBrowser"
```
Comment (aerozen3) [inline packages/gestalt/src/__snapshots__/Avatar.test.js.snap]: hmm not crazy that we're introducing this classname in all the tests, maybe there's another way?  Ultimate I need to turn off `will-change: transform` for Safari full screen, but there's not an easy way to use css selectors for Safari: https://stackoverflow.com/questions/16348489/is-there-a-css-hack-for-safari-only-not-chrome.  And the element that is full screen is a child of `Mask`

Comment (btraut) [reply packages/gestalt/src/__snapshots__/Avatar.test.js.snap]: It's totally fine that it's added to all the snapshots.

File: packages/gestalt/src/Mask.css [hunk 2]
```diff
@@ -1,7 +1,9 @@
 .Mask {
   composes: overflowHidden from "./Layout.css";
   composes: relative from "./Layout.css";
+}
 
+.hintBrowser {
```
Comment (pfarejowicz) [inline packages/gestalt/src/Mask.css]: +1 to @btraut comment, this naming feels off

**Comment by aerozen3** [comment]
I got the name looking at the example in MDN docs here: https://developer.mozilla.org/en-US/docs/Web/CSS/will-change, but thinking about it some more, I agree `willChangeTransform` is probably clearer, updating it

**Comment by aerozen3** [comment]
added to the change log and docs:
Screen Shot 2019-03-12 at 4 50 43 PM

**Pull request merged**